### PR TITLE
docs: fix explanation pages

### PR DESCRIPTION
**Pull request by linostar**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

This PR is part of series of proofreading Rockcraft and Craft Parts documentation, making it up-to-date and conforming to standards, and adding/fixing sections and links where necessary.

### CODE REVIEW

**Comment by linostar** [comment]
@cjdcordeiro FYI.

**Comment by sergiusens** [comment]
```
/home/docs/checkouts/readthedocs.org/user_builds/canonical-craft-parts/checkouts/765/docs/common/craft-parts/explanation/how_parts_are_built.rst:39: WARNING: undefined label: 'overlays'
/home/docs/checkouts/readthedocs.org/user_builds/canonical-craft-parts/checkouts/765/docs/common/craft-parts/explanation/how_parts_are_built.rst:39: WARNING: undefined label: 'overlays'
```

**Comment by cjdcordeiro** [comment]
are the failing tests related?

**Comment by linostar** [comment]
> are the failing tests related?

The failing tests seem not be related. They are dpkg lock errors when running `apt install` commands in tests.

**Review by tigarmo — APPROVED** [review]

File: docs/common/craft-parts/explanation/parts.rst [hunk 1]
```diff
@@ -165,12 +159,18 @@ system packages that need to be deployed with the part are specified using
 the :ref:`stage_snaps` and :ref:`stage_packages` properties. Files to be
 deployed are specified using the :ref:`stage` property.
 
+.. ifconfig:: project in ("Rockcraft",)
```
Comment (tigarmo) [inline docs/common/craft-parts/explanation/parts.rst:162]: This shouldn't be scoped to Rockcraft - Snapcraft also supports chisel slices I believe

File: docs/common/craft-parts/explanation/overlay_parameters.rst [hunk 2]
```diff
@@ -4,13 +4,13 @@ Overlay Parameters
 ------------------
 
 A part has three parameters that can be used to adjust how the overlay step
-works: ``overlay-packages``, ``overlay-script`` and ``overlay-filter``.
```
Comment (cjdcordeiro) [inline docs/common/craft-parts/explanation/overlay_parameters.rst:7]: I don't see any current usages of `overlay-filter` in our org, but if you believe that was ever the case, I'd probably suggest adding some sort of note about this change

File: docs/common/craft-parts/explanation/overlays.rst [hunk 3]
```diff
@@ -1,6 +1,8 @@
-****************
-``OVERLAY`` Step
-****************
+.. _overlays:
+
+************
+OVERLAY Step
```
Comment (cjdcordeiro) [inline docs/common/craft-parts/explanation/overlays.rst:4]: Does it need to be upper case? In https://documentation.ubuntu.com/rockcraft/en/stable/explanation/overlay-step/  we have it sentence case

File: docs/common/craft-parts/explanation/parts.rst [hunk 4]
```diff
@@ -165,12 +159,18 @@ system packages that need to be deployed with the part are specified using
...
+
+   Chisel slices can be specified in :ref:`stage_packages` property as well, but
```
Comment (cjdcordeiro) [inline docs/common/craft-parts/explanation/parts.rst:164]:
```suggestion
   Chisel slices can be specified in :ref:`stage_packages` as well, but
```

File: docs/common/craft-parts/explanation/parts.rst [hunk 5]
```diff
@@ -165,12 +159,18 @@ system packages that need to be deployed with the part are specified using
...
+   they are mutually exclusive to use with deb system packages.
```
Comment (cjdcordeiro) [inline docs/common/craft-parts/explanation/parts.rst:165]:
```suggestion
   they cannot be mixed with deb packages. 
```

File: docs/common/craft-parts/reference/part_properties.rst [hunk 6]
```diff
@@ -69,7 +69,10 @@ build-snaps
 
 The snaps to be installed in the build environment before the build is
 performed. These provide libraries and executables that the part needs during
-the build process.
+the build process. They take the form of ``<snap>/<version>/<channel>``,
+``<snap>/<channel>`` or just ``<snap>``. For example, the ``juju`` snap could be
+specified as ``juju/2.9/stable``, ``juju/latest/stable`` or ``juju/stable`` to
```
Comment (cjdcordeiro) [inline docs/common/craft-parts/reference/part_properties.rst:74]: Why the `juju` example? It seems like an uncommon snap to use as an example.

File: docs/common/craft-parts/explanation/overlay_parameters.rst (hunk 2 above)
Comment (linostar) [reply docs/common/craft-parts/explanation/overlay_parameters.rst:7]: Inspecting the git history, it does seem this was a an old typo, as the field name is "overlay-files" and its alias is "overlay". I am going to put it as "overlay" in the docs, as aliases are more preferrable to use.

File: docs/common/craft-parts/reference/part_properties.rst (hunk 6 above)
Comment (linostar) [reply docs/common/craft-parts/reference/part_properties.rst]: I changed it to `node`.

**Review by cjdcordeiro — APPROVED** [review]
LGTM if the tests are unrelated

**Comment by linostar** [comment]
> LGTM if the tests are unrelated

They are unrelated.
That being said, it is not clear to me what is causing those dpkg lock errors.

**Review by lengau — APPROVED** [review]

**Pull request merged**